lib/compiler.js

```diff
@@ -91,12 +97,41 @@ module.exports = function compile(options) {
     models: modelInstructions,
     middleware: middlewareInstructions,
     components: componentInstructions,
+    mixins: mixinInstructions,
     files: {
       boot: bootScripts
     }
   });
 };
 
+function buildAllMixinInstructions(appRootDir, mixinSources, options) {
+  var extensions = _.without(_.keys(require.extensions), '.json');
+  var files = options.mixins || [];
+  mixinSources.forEach(function(dir) {
+    dir = path.resolve(appRootDir, dir);
+    files = files.concat(findScripts(dir, extensions));
+  });
+
+  var mixins = {};
+  files.forEach(function(filepath) {
+    var dir = path.dirname(filepath);
+    var ext = path.extname(filepath);
+    var name = path.basename(filepath, ext);
+    var metafile = path.join(dir, name + '.json');
+    name = normalizeMixinName(name, options);
+    var meta = {};
+    meta.name = name;
+    if (fs.existsSync(metafile)) {
+      // May overwrite name, not id or filepath
+      _.extend(meta, require(metafile));
+    }
+    meta.filepath = filepath;
+    mixins[name] = meta;
```

Comment:
The key in `mixins` may be different from the `name` provided in the json file. Is this intentional?

I think it will be better to either use `meta.name` as the key, or else register the mixin under both names (the one based on a file name, and the one provided by the json file).